src/NLog/Targets/AnsiConsoleColor.cs

```diff
@@ -0,0 +1,136 @@
+    
+    internal static class AnsiConsoleColor
+    {
+        internal static string GetForegroundColorEscapeCode(ConsoleColor color)
+        {
+            switch (color)
+            {
+                case ConsoleColor.Black:
+                    return "\x1B[30m";
+                case ConsoleColor.Blue:
+                    return "\x1B[94m";
+                case ConsoleColor.Cyan:
+                    return "\x1B[96m";
+                case ConsoleColor.DarkBlue:
+                    return "\x1B[34m";
+                case ConsoleColor.DarkCyan:
+                    return "\x1B[36m";
+                case ConsoleColor.DarkGray:
+                    return "\x1B[90m";
+                case ConsoleColor.DarkGreen:
+                    return "\x1B[32m";    
+                case ConsoleColor.DarkMagenta:
+                    return "\x1B[35m";
+                case ConsoleColor.DarkRed:
+                    return "\x1B[31m";
+                case ConsoleColor.DarkYellow:
+                    return "\x1B[33m";
+                case ConsoleColor.Gray:
+                    return "\x1B[37m";
+                case ConsoleColor.Green:
+                    return "\x1B[92m";
+                case ConsoleColor.Magenta:
+                    return "\x1B[95m";
+                case ConsoleColor.Red:
+                    return "\x1B[91m";
+                case ConsoleColor.White:
+                    return "\x1B[97m";
+                case ConsoleColor.Yellow:
+                    return "\x1B[93m";
+                default:
+                    return TerminalDefaultForegroundColorEscapeCode; // default foreground color
+            }
+        }
+        
+        internal static string TerminalDefaultForegroundColorEscapeCode
+        {
+            get { return "\x1B[39m"; }
+        }
+
+        internal static string GetBackgroundColorEscapeCode(ConsoleColor color)
+        {
+            switch (color)
+            {
+                case ConsoleColor.Black:
+                    return "\x1B[40m";
+                case ConsoleColor.Blue:
+                    return "\x1B[104m";
+                case ConsoleColor.Cyan:
+                    return "\x1B[106m";
+                case ConsoleColor.DarkBlue:
+                    return "\x1B[44m";
+                case ConsoleColor.DarkCyan:
+                    return "\x1B[46m";
+                case ConsoleColor.DarkGray:
+                    return "\x1B[100m";
+                case ConsoleColor.DarkGreen:
+                    return "\x1B[42m";    
+                case ConsoleColor.DarkMagenta:
+                    return "\x1B[45m";
+                case ConsoleColor.DarkRed:
+                    return "\x1B[41m";
+                case ConsoleColor.DarkYellow:
+                    return "\x1B[43m";
+                case ConsoleColor.Gray:
+                    return "\x1B[47m";
+                case ConsoleColor.Green:
+                    return "\x1B[102m";    
```

Comment:
As far as I can see the bright background colors are not part of the original specifications. I guess I should just document that the bright colors might not be supported on older terminals.